src/app/client/src/app/modules/core/services/copy-content/copy-content.service.ts

```diff
@@ -115,36 +116,52 @@ export class CopyContentService {
       return response;
     }));
   }
-
   /**
    * This method prepares the request body for the copy API
    * @param {contentData} ContentData Content data which will be copied
+   * @description  request will be formed based on the mimetype
    */
   formatData(contentData: ContentData) {
     const userData = this.userService.userProfile;
+    let req;
     if (contentData.description === undefined) {
       contentData.description = '';
     }
     let creator = userData.firstName;
     if (!_.isEmpty(userData.lastName)) {
       creator = userData.firstName + ' ' + userData.lastName;
     }
-    const req = {
-      request: {
-        content: {
-          name: 'Copy of ' + contentData.name,
-          description: contentData.description,
-          code: contentData.code + '.copy',
-          creator: creator,
-          createdFor: userData.organisationIds,
-          createdBy: userData.userId,
-          organisation: _.uniq(this.userService.orgNames),
-          framework: '',
-          mimeType: contentData.mimeType,
-          contentType: contentData.contentType
+    if (_.get(contentData,'mimeType') === 'application/vnd.sunbird.questionset') {
```

Comment:
Why are we written separate if else  for questionset. 
Will the questionset copy fails if we send the same request that we are passing?

Reply:
Request metadata is differ, 
 **question set request body** ::  req = {request: { questionset: {... } }  and for question-set content type metadata is not required 
**content request body**  req = { request: { content: { ...}} .

Reply:
The same properties present in the Content request as well right. Can't I create the content with same properties of Question-Set?

Reply:
@vinukumar-vs  
I agree, but according to the backend implementation  -> http://941.qa.docs.sunbird.org/4.9.0/apis/questionapi//#operation/Import%20QuestionSet
backend is expecting **request.questionset** in request body so, I have implemented the same.

CC: @rajeevsathish @rajeshkumaravel